datafusion/src/row/writer.rs

```diff
@@ -17,19 +17,24 @@
 
 //! Reusable row writer backed by Vec<u8> to stitch attributes together
 
+#[cfg(feature = "jit")]
 use crate::error::Result;
 #[cfg(feature = "jit")]
 use crate::reg_fn;
 #[cfg(feature = "jit")]
 use crate::row::fn_name;
-use crate::row::{estimate_row_width, fixed_size, get_offsets, supported};
+use crate::row::{
+    estimate_row_width, fixed_size, get_offsets, schema_null_free, supported,
+};
 use arrow::array::*;
 use arrow::datatypes::{DataType, Schema};
 use arrow::record_batch::RecordBatch;
 use arrow::util::bit_util::{ceil, round_upto_power_of_2, set_bit_raw, unset_bit_raw};
+#[cfg(feature = "jit")]
```

Comment:
I think over time it would be good to start trying to encapsulate the JIT'd code more (as in reduce the number of `#[cfg(feature = "jit")]` calls -- perhaps by defining a common interface for creating jit and non jit versions. As I am interested in getting more involved in this project, I would be happy to try and do so (or do it as part of a larger body of work)

Reply:
Ah, that would be great! Thanks for the offering.

Reply:
I'll see what I can do over the next day or two